.github/actions/setup_gcc_ubuntu/action.yml

```diff
@@ -76,6 +70,9 @@ runs:
     - name: Install gcc
       shell: bash
       run: |
+        if [[ ${{ inputs.gcc-version == 'gcc-7' }} ]]; then
+          echo "deb http://archive.ubuntu.com/ubuntu focal main universe" >> "/etc/apt/sources.list"
```

Comment:
Using `apt-add-repository "deb http://archive.ubuntu.com/ubuntu focal main universe"` should do exactly the same thing and would be consistent with above, but not that important

Reply:
Good shout.  Also, combining it with the above (just chucking universe onto the end of the original command) works.  So that's what we're going with